staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -144,6 +147,38 @@ func GetPrefixTransformers(config *ResourceConfig) ([]value.PrefixTransformer, e
 			found = true
 		}
 
+		if provider.KMS != nil {
+			if found == true {
+				return nil, fmt.Errorf("more than one provider specified in a single element, should split into different list elements")
+			}
+			f, err := os.Open(provider.KMS.ConfigFile)
```

Comment:
An approach which I've proposed before, and I've seen used in the cloudproviders package:
Have a field `config map[string]interface{}` on the `KMSConfig` struct. Send it to the plugin factory. The plugin factory would use the `mapstructure` library to parse it into a struct safely.

So this way you can support both `config:` and `configfile:` attributes on the `kms` configuration. Just putting this out here.